### PR TITLE
8264672: runtime/ParallelLoad/ParallelSuperTest.java timed out

### PR DESCRIPTION
**Pull request by coleenp**
Added a flag so that thread 2 will notify thread 1 after it's waiting, and not before.
Ran 100x on macosx-x64-debug without timeout (actually ran original 100x without timeout also).
The test relies on one thread waiting at a place where another can find the class loading in progress.  Alternate suggestions welcome.
Thanks!
Coleen

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8264672](https://bugs.openjdk.java.net/browse/JDK-8264672): runtime/ParallelLoad/ParallelSuperTest.java timed out

### Reviewers
 * [Harold Seigel](https://openjdk.java.net/census#hseigel) (@hseigel - **Reviewer**) ⚠️ Review applies to 5958bdcb3923cc671f14f2225fa353ff2c5d4369
 * [Lois Foltan](https://openjdk.java.net/census#lfoltan) (@lfoltan - **Reviewer**) ⚠️ Review applies to d5991d9ec5203f68fae9787db37770b051ecda49
 * [David Holmes](https://openjdk.java.net/census#dholmes) (@dholmes-ora - **Reviewer**) ⚠️ Review applies to 5958bdcb3923cc671f14f2225fa353ff2c5d4369

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/3375/head:pull/3375` \
`$ git checkout pull/3375`

Update a local copy of the PR: \
`$ git checkout pull/3375` \
`$ git pull https://git.openjdk.java.net/jdk pull/3375/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 3375`

View PR using the GUI difftool: \
`$ git pr show -t 3375`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/3375.diff">https://git.openjdk.java.net/jdk/pull/3375.diff</a>

</details>

### CODE REVIEW

**Review by hseigel — COMMENTED** [review]
Changes look good.  One question: does 'ready' need to be volatile ?

**Review by lfoltan — APPROVED** [review]
LGTM.
Lois

**Review by hseigel — APPROVED** [review]
LGTM!

File: test/hotspot/jtreg/runtime/ParallelLoad/MyLoader.java [hunk 1]
```diff
@@ -61,6 +61,7 @@ public Class loadClass(String name) throws ClassNotFoundException {
     private static boolean parallel = false;
     private Object sync = new Object();
     private Object thread_sync = new Object();
+    private static volatile boolean ready = false;
```
Comment (dholmes-ora) [inline test/hotspot/jtreg/runtime/ParallelLoad/MyLoader.java:64]: The name "waiting" would be more appropriate.

**Review by dholmes-ora — APPROVED** [review]
Hi Coleen,

The synchronization in this test is highly suspect, but your change certainly helps address one of the race conditions.

Thanks,
David

File: test/hotspot/jtreg/runtime/ParallelLoad/MyLoader.java (hunk 1 above)
Comment (coleenp) [reply test/hotspot/jtreg/runtime/ParallelLoad/MyLoader.java:64]: Ok.

**Pull request closed without merging**